lib/virt_feature_test_base.pm

```diff
@@ -157,7 +158,21 @@ sub post_fail_hook {
     my ($self) = shift;
 
     $self->{"stop_run"} = time();
-    assert_script_run("history -w /root/commands_history");
+    unless (defined(script_run("history -w /root/commands_history", die_on_timeout => 0))) {
+        # Switch to sol console to check serial console output
+        # It is useful in the case of host crash and reboot
+        record_info("WARN", "ssh connection is broken and switch to SOL console", result => 'softfail');
+        select_console 'sol', await_console => 0;
+        save_screenshot;
+        wait_still_screen 5;
+        check_screen("linux-login", 120);
+        reset_consoles;
+        record_info("WARN", "switch back to ssh console to collect logs", result => 'softfail');
+        select_console('root-ssh');
+        script_run("uptime");
+        script_run("ls -l /var/crash/");
```

Comment:
If you want to have more info about crash, I think you can also print dmesg out if there is a crash.

Reply:
I would like to check if a reboot or a crash has happened. More detailed info would be uploaded by our logging system after ssh re-connection.